### ISSUE TITLE
Cannot choose language in "xournalpp-1.1.0+dev-nightly.20210221"

### ISSUE THREAD

**Issue by MichelReims** [xournalpp/xournalpp#2844]
OS : Windows 10
Version of xournalpp : xournalpp-1.1.0+dev-nightly.20210221

Issue : It is no longer possible to choose the language when it was possible in previous versions

**Comment by rolandlo** [comment]
@MichelReims Can you say what was the last version where you could choose the GUI language?

**Comment by MichelReims** [comment]
For example I can choose language in "xournalpp-1.1.0+dev-nightly.20210204-windows", but not in "xournalpp-1.1.0+dev-nightly.20210207-windows".
I think the problem appeared when during the installation of Xournalpp there is a new dialog box : "Install for all users"

**Comment by rolandlo** [comment]
Ok, that's when we overhauled the Windows installer. Probably the location of the localization files is not found. I will take a look how to fix it.

**Comment by MichelReims** [comment]
Thank you

**Comment by rolandlo** [comment]
My observation (under Windows):
- If Xournal++ from the Start Menu localizations work normally.
- If Xournal++ is run from the Command Prompt from the `bin` subfolder of the installation folder (typing `xournalpp.exe`) it runs fine as well
- If Xournal++ is run from the Command Prompt from a different folder (e.g. from the installation folder using `bin\xournalpp.exe`) localizations are not found
- If Xournal++ is run via the context menu of a .xopp-file in the file manager localizations are not found.

The reason is simple: Xournal++ will search the localizations either 
1) from the path specified in the TEXTDOMAINDIR environment variable (which we do not set in Windows; it is used for the AppImage) or 
2) relative to the present working directory in `..\share\locale`. The present working directory should be the folder which contains `xournalpp.exe`, otherwise this won't work.

@MichelReims Can you please verify my findings so far?

**Comment by rolandlo** [comment]
One more observation:
If you create an environment variable TEXTDOMAINDIR with value `/c/Program Files/Xournal++/share/locale` then the GUI picks up the system language, but you cannot change the language in the preferences

**Comment by MichelReims** [comment]
On my computer, the only time xournalpp works normally is when I run it from "bin" subfolder of the installation folder. In other cases it does not work normally.
It's already good like that

**Comment by rolandlo** [comment]
Did you try to uninstall Xournal++ before reinstalling the new version? Maybe the start menu entry is from an older version.

**Comment by MichelReims** [comment]
Yes I uninstalled and reinstalled it. It does not change anything.
The environment variable doesn't change anything either

**Comment by rolandlo** [comment]
I can now see why it doesn't work on Windows to use the TEXTDOMAINDIR environment variable. The output
```
TEXTDOMAINDIR = C:/Program Files/Xournal++/share/locale, PACKAGE_LOCALE_DIR = ../share/locale/, chosen directory = C
``` 
reveals it all. Namely the `:` is considered as a separator between two directories (which it is not here)
```` c++
    // Only consider first path in environment variable
    std::string directories;
    if (gettextEnv) {
        directories = std::string(gettextEnv);
        size_t firstDot = directories.find(':');
        if (firstDot != std::string::npos) {
            directories = directories.substr(0, firstDot);
        }
    }
```

**Comment by rolandlo** [comment]
With PR #2845 merged the TEXTDOMAINDIR environment variable (which you can set like any environment variable as described [here](https://www.computerhope.com/issues/ch000549.htm#windows10)) can be used to set the path for the localizations, so you can open a .xopp-file from the context menu with working localization. The value of this environment variable should be `/c/Program Files/Xournal++/share/locale` assuming default installation folder.

There are now 2 ways of getting correct localizations:
1) Run Xournal++ from the `bin` folder
2) Use TEXTDOMAINDIR environment variable.I

So I will close this issue.

**Comment by rolandlo** [comment]
We should probably make the Start menu entry from the bin subfolder.

![grafik](https://user-images.githubusercontent.com/40485433/124910069-75b46f80-dfeb-11eb-9a2b-bdd3c7521139.png)

**Comment by rolandlo** [comment]
I'm closing the issue again in view of the fix #3215
